terraform-modules/k8s/k8s-node-pool.tf

```diff
@@ -3,45 +3,44 @@
 */
 
 resource "google_container_node_pool" "node-pool" {
-  provider            = "google"
-  depends_on  = [
-    "google_container_cluster.cluster"
-  ]
-  name        = "${var.cluster_name}-np"
-  zone        = "${var.zone}"
-  cluster     = "${google_container_cluster.cluster.name}"
-  node_count  = "${var.node_pool_count}"
+  provider   = google
+  depends_on = [google_container_cluster.cluster]
+  name       = "${var.cluster_name}-np"
+  zone       = var.zone
+  cluster    = google_container_cluster.cluster.name
+  node_count = var.node_pool_count
 
   management {
     # CIS compliance: enable automatic repair
     auto_repair = true
+
     # CIS compliance: enable automatic upgrade
     auto_upgrade = true
   }
 
-  version = "${var.node_version}"
+  version = var.node_version
 
   node_config {
     # CIS compliance: COS image
     image_type      = "COS"
-    machine_type    = "${var.node_pool_machine_type}"
-    disk_size_gb    = "${var.node_pool_disk_size_gb}"
-    service_account = "${var.node_service_account}"
+    machine_type    = var.node_pool_machine_type
+    disk_size_gb    = var.node_pool_disk_size_gb
+    service_account = var.node_service_account
 
-    metadata {
-      google-compute-enable-virtio-rng = "${var.enable_node_rng}"
-      disable-legacy-endpoints = "true"
+    metadata = {
+      google-compute-enable-virtio-rng = var.enable_node_rng
```

Comment:
@RLuckom what does this label do? When I `terraform plan` against my existing k8s cluster in dev, I see:
1. The old default value of `"false"` for the `var.enable_node_rng` somehow got translated to `"0"`
2. Now the value of `false` isn't being translated, so the metadata value will change from `"0" -> "false"`
3. That change will force the whole node pool to rebuild

As far as I can tell from the `plan` output, this is the only change that forces a full teardown & redeploy in the whole upgrade. I want to be sure it's worth it.

Reply:
I think that if possible it would be better to allow the module client to supply the entire `metadata` block, and if that's ok (as a future improvement, not part of this PR) then I don't really care what we do here. It sounds like if you made that change now it might let you not force a redeploy, so if that's better for you go ahead.